app/src/app/api/v0/inventory/[inventory]/preferences/route.ts

```diff
@@ -0,0 +1,321 @@
+  const inventory = await UserService.findUserInventory(
+    params.inventory,
+    session,
+    [
+      {
+        model: db.models.InventoryValue,
+        as: "inventoryValues",
+        include: [
+          {
+            model: db.models.ActivityValue,
+            as: "activityValues",
+            include: [
+              {
+                model: db.models.GasValue,
+                as: "gasValues",
+                include: [
+                  { model: db.models.EmissionsFactor, as: "emissionsFactor" },
+                ],
+              },
+            ],
+          },
+          {
+            model: db.models.GasValue,
+            as: "gasValues",
+          },
+          {
+            model: db.models.DataSource,
+            attributes: ["datasourceId", "sourceType"],
+            as: "dataSource",
+          },
+        ],
+      },
+    ],
+  );
+  if (!inventory.year) {
+    throw new createHttpError.BadRequest(
+      `Inventory ${inventory.inventoryId} is missing a year number`,
+    );
+  }
+  const MAX_YEARS_DIFFERENCE = 10;
+  const populationEntries = await db.models.Population.findAll({
+    attributes: ["year", "population"],
+    where: {
+      cityId: inventory.cityId,
+      year: {
+        [Op.gte]: inventory.year - MAX_YEARS_DIFFERENCE,
+        [Op.lte]: inventory.year + MAX_YEARS_DIFFERENCE,
+      },
+      population: {
+        [Op.ne]: null,
+      },
+    },
+    order: [["year", "DESC"]],
+  });
+
+  const population = findClosestYearToInventory(
+    populationEntries as PopulationEntry[],
+    inventory.year,
+    MAX_YEARS_DIFFERENCE,
+  );
+  if (!population) {
+    throw new createHttpError.NotFound(
+      `Population data not found for city ${inventory.cityId} for year ${inventory.year}`,
+    );
+  }
+
+  const output: InventoryResponse = inventory.toJSON();
+  output.city.populationYear = population.year;
+  output.city.population = population.population || 0;
+  let body: Buffer | null = null;
+  let headers: Record<string, string> | null = null;
+
+  switch (req.nextUrl.searchParams.get("format")?.toLowerCase()) {
+    case "csv":
+      body = await inventoryCSV(inventory);
+      headers = {
+        "Content-Type": "text/csv",
+        "Content-Disposition": `attachment; filename="inventory-${inventory.city.locode}-${inventory.year}.csv"`,
+      };
+      break;
+    case "xls":
+      body = await inventoryXLS(inventory);
+      headers = {
+        "Content-Type": "application/vnd.ms-excel",
+        "Content-Disposition": `attachment; filename="inventory-${inventory.city.locode}-${inventory.year}.xls"`,
+      };
+      break;
+    case "json":
+    default:
+      body = Buffer.from(JSON.stringify({ data: output }), "utf-8");
+      headers = {
+        "Content-Type": "application/json",
+      };
+      break;
+  }
+
+  return new NextResponse(body, { headers });
+});
+
+async function inventoryCSV(inventory: Inventory): Promise<Buffer> {
```

Comment:
This function should not be in this file. It's supposed to be in the download route, not sure why it and the excel functionality etc. was copied here...

Reply:
sorry, removed this file until we implement the backend!